src/Storage/StreamWrapper.php

```diff
@@ -0,0 +1,271 @@
+        if ($this->context) {
+            $options = stream_context_get_options($this->context);
+            if (array_key_exists($this->protocol, $options)) {
+                $this->options = $options[$this->protocol] ?: [];
+            }
+        }
+        return $this->options;
+    }
+
+    private function getStream()
+    {
+        return $this->stream;
+    }
+
+    private function isWriteable()
+    {
+        return in_array($this->mode, ['w', 'wb', 'wt']);
+    }
+
+    private function isReadable()
+    {
+        return in_array($this->mode, ['r', 'rb', 'rt']);
+    }
+
+    // @codingStandardsIgnoreStart
+    /**
+     * Callback handler for when we try to read a certain number of bytes.
+     *
+     * @param int $count The number of bytes to read.
+     *
+     * @return string
+     */
+    public function stream_read($count)
+    {
+        // @codingStandardsIgnoreEnd
+        return $this->getStream()->read($count);
+    }
+
+    // @codingStandardsIgnoreStart
+    /**
+     * Callback handler for when we try to write data to the stream.
+     *
+     * @param string|stream $data The data to write
+     *
+     * @return int The number of bytes written.
+     */
+    public function stream_write($data)
+    {
+        // @codingStandardsIgnoreEnd
+        return $this->getStream()->write($data);
+    }
+
+    // @codingStandardsIgnoreStart
+    /**
+     * Callback handler for getting data about the stream.
+     *
+     * @return array
+     */
+    public function stream_stat()
+    {
+        // @codingStandardsIgnoreEnd
+        return [
+            'dev'     => 0,
+            'ino'     => 0,
+            'mode'    => $this->isWriteable() ? 33188 : 33060, // equivalent to 10644 and 10444 in octal
+            'nlink'   => 0,
+            'uid'     => 0,
+            'gid'     => 0,
+            'rdev'    => 0,
+            'size'    => $this->getStream()->getSize(),
+            'atime'   => 0,
+            'mtime'   => 0,
+            'ctime'   => 0,
+            'blksize' => 0,
+            'blocks'  => 0
+        ];
+    }
+
+    // @codingStandardsIgnoreStart
+    /**
+     * Callback handler for checking to see if the stream is at the end of file.
+     *
+     * @return bool
+     */
+    public function stream_eof()
+    {
+        // @codingStandardsIgnoreEnd
+        return $this->getStream()->eof();
+    }
+
+    // @codingStandardsIgnoreStart
+    /**
+     * Callback handler for trying to close the stream.
+     */
+    public function stream_close()
+    {
+        // @codingStandardsIgnoreEnd
+        if (isset($this->stream)) {
+            $this->getStream()->close();
+        }
+    }
+
+    // @codingStandardsIgnoreStart
+    /**
+     * Callback handler for trying to seek to a certain location in the stream.
+     */
+    public function stream_seek(int $offset, int $whence)
+    {
+        // @codingStandardsIgnoreEnd
+        return $this->getStream()->seek($offset, $whence);
```

Comment:
`stream_seek` should return a bool on success/failure. We might need to do a little more to wrap this call here, as it throws an exception on failure or returns nothing on success - [Stream.php](https://github.com/guzzle/psr7/blob/master/src/Stream.php#L195)